sdk/test-utils/perf/src/managerProgram.ts

```diff
@@ -116,8 +116,20 @@ export class ManagerPerfProgram implements PerfProgram {
   }
 
   private createWorkers(): void {
-    const cpuOption = this.parsedOptions.cpus.value ?? 0;
+    let cpuOption = this.parsedOptions.cpus.value ?? 0;
     const parallels = this.parsedOptions.parallel.value ?? 1;
+
+    if (this.parsedOptions.profile.value) {
+      if (this.parsedOptions.cpus.value !== 1) {
```

Comment:
If `--cpus` is set and `--profile` is enabled, I would print a warning and override CPUs to 1, rather than throwing an error.  This way, users can always safely append `--profile` to an existing command line, even if it includes `--cpus`.

Reply:
That sounds good to me as long as the warning is obvious in CI output traces etc. I think I lean towards throwing the error just because it's trivial to figure out how to just fix the invocation without having an ignored `--cpus` param. However, if we go that route, we should only throw if `--cpus` is explicitly set. This code seems like it would throw even if `--cpus` is unset, and in that case we should default to 1.

Reply:
I definitely agree `--profile` needs to work if `--cpus` is unset.  I'm OK with either throwing or warning if both are set, though I would lean to just warning.

Reply:
Updated to throw when profile=true and cpus is set to something other than 1.